### PR TITLE
Added XenServer support

### PR DESCRIPTION
**Pull request by kongslund**

### CODE REVIEW

**Comment by kongslund** [comment]
Ping: @josephreynolds since I noticed that you have an empty project named `chef-provisioning-xenserver`. I think this PR would be useful to you.

File: lib/chef/provisioning/fog_driver/providers/xenserver.rb [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            new_compute_options[:xenserver_timeout] ||= 300
+            new_compute_options[:xenserver_redirect_to_master] ||= true
+
+            id = result[:driver_options][:compute_options][:xenserver_url]
+
+            [result, id]
+          end
+
+          def server_for(machine_spec)
+            if machine_spec.location
```
Comment (tyler-ball) [inline lib/chef/provisioning/fog_driver/providers/xenserver.rb]: This should be `machine_spec.reference`

File: lib/chef/provisioning/fog_driver/providers/xenserver.rb [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+                result[machine_spec] = compute.servers.get(compute.get_by_uuid(machine_spec.location['server_id'], 'VM'))
+              else
+                result[machine_spec] = nil
+              end
+            end
+            result
+          end
+          
+          def create_many_servers(num_servers, bootstrap_options, parallelizer)
+            parallelizer.parallelize(1.upto(num_servers)) do |i|
```
Comment (tyler-ball) [inline lib/chef/provisioning/fog_driver/providers/xenserver.rb]: An issue users have had in the past is hitting API limits.  If that will be an issue accessing the XenServer provisioner, could you change `num_servers` to `[num_servers, 5].min`?

File: lib/chef/provisioning/fog_driver/providers/xenserver.rb [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+      end
+    end
+
+  end
+end
+
+module Fog
+  class Logger
+    def self.deprecation(message)
+      # Silence...ahh
```
Comment (tyler-ball) [inline lib/chef/provisioning/fog_driver/providers/xenserver.rb]: Sooooooooo... I understand the desire to do this, but its probably a bad idea in the long run.  Could we perhaps redirect them to `Chef::Log.debug` with a short header?

**Comment by tyler-ball** [comment]
Other than a few comments this LGTM.  I don't have a XenServer that I can test this code against - has it worked correctly for you?

**Comment by kongslund** [comment]
Hi @tyler-ball. Thanks for the feedback. I will look into incorporating it. However, it may take a little while before I have an opportunity to do so, so please be patient. :-)

File: lib/chef/provisioning/fog_driver/providers/xenserver.rb (hunk 3 above)
Comment (kongslund) [inline lib/chef/provisioning/fog_driver/providers/xenserver.rb]: You mean something like `Chef::Log.debug('Fog: ' + message)`?

**Comment by kongslund** [comment]
Hi @tyler-ball. Your suggestions have been incorporated and tested.

File: lib/chef/provisioning/fog_driver/providers/xenserver.rb (hunk 2 above)
Comment (kongslund) [inline lib/chef/provisioning/fog_driver/providers/xenserver.rb]: There are no API limits.

File: lib/chef/provisioning/fog_driver/providers/xenserver.rb (hunk 3 above)
Comment (tyler-ball) [inline lib/chef/provisioning/fog_driver/providers/xenserver.rb]: Exactly!  Thanks

**Pull request merged**